WordPress/src/main/java/org/wordpress/android/ui/WPWebViewActivity.java

```diff
@@ -138,7 +150,10 @@ public static void openUrlByUsingBlogCredentials(Context context, SiteModel site
         intent.putExtra(WPWebViewActivity.DISABLE_LINKS_ON_PAGE, true);
             intent.putExtra(ALLOWED_URLS, listOfAllowedURLs);
         if (post != null) {
-            intent.putExtra(WPWebViewActivity.SHARABLE_URL, WPMeShortlinks.getPostShortlink(site, post));
```

Comment:
Please note that, the reason for this change is that `WPMeShortlinks.getPostShortlink` will always return `null` for self-hosted sites due to the below code. So, we should just share the post's url, otherwise the webview's current url would be used.

```
if (!SiteUtils.isAccessibleViaWPComAPI(site)) {
    return null;
}
```